build.ps1

```diff
@@ -84,124 +86,121 @@ Function CheckPublishTools()
 
 Function CheckTools($commands)
 {
-    foreach($command in $commands)
+    foreach ($command in $commands)
     {
         $info = Get-Command $command -ErrorAction SilentlyContinue
-        if ($info -eq $null)
+        if (-not $info)
         {
             throw "Toolset not found: '$command' is missing."
         }
     }
 }
 
-Function BuildProject($path, $message) {
-
+Function BuildProject($path, $message)
+{
     $label = "BUILD: --- $message $configuration ---"
 
     Write-Host
     Write-Host -ForegroundColor Cyan $label
-    cd (Join-Path $rootDir $path)
 
-    if ($clean) {
-        & dotnet clean --verbosity $verbosity --configuration $configuration
-        if ($LASTEXITCODE -ne 0) {
+    $projectPath = Join-Path $rootDir $path
+
+    if ($clean)
+    {
+        & dotnet clean $projectPath --verbosity $verbosity --configuration $configuration
+        if ($LASTEXITCODE -ne 0)
+        {
             throw "Clean failed: $label"
         }
     }
 
-    & dotnet build --verbosity $verbosity --configuration $configuration
+    & dotnet build $projectPath --verbosity $verbosity --configuration $configuration
```

Comment:
By specifying the project path, we don't need to change to the directory (line 103 removed)